examples/autoqasm/3_Iterative_phase_estimation.ipynb

```diff
@@ -0,0 +1,214 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "08d77a63",
+   "metadata": {},
+   "source": [
+    "# Iterative phase estimation\n",
+    "In this notebook, we implement the iterative phase estimation (IPE) algorithm described in Ref [1] and [2]. In the algorithm, two qubits are used, with the 0th qubit being the ancilla qubit and the 1st qubit being the data qubit. The phase oracle in this example uses an ancilla qubit to add a phase $e^{i\\phi}$ to the $|1\\rangle$ component of the state of the data qubit. The oracle is written as a subroutine that will be used by the quantum program. "
```

Comment:
I think I may have had some comments on this paragraph in the original PR that weren't addressed. Mainly about contextualizing IPE